store/src/java/com/zimbra/cs/service/UserServlet.java

```diff
@@ -907,7 +907,7 @@ private void resolveFormatter(UserServletContext context) throws UserServletExce
                 try {
                     Provisioning prov = Provisioning.getInstance();
                     Server server = prov.getServer(context.getAuthAccount());
-                    if (!StringUtil.isNullOrEmpty(server.getDocumentServerHost())) {
+                    if (server != null && !StringUtil.isNullOrEmpty(server.getDocumentServerHost())) {
```

Comment:
in which case, we will not get server object?

Reply:
and why publicly sharing a document is calling this code related to preview of documents?

Reply:
when there is no auth account

Reply:
> and why publicly sharing a document is calling this code related to preview of documents?

this is not just code to preview document, its the code path for getting the document too which is common for all documents fetch